### PR TITLE
Add support to use the automate chef server

### PR DESCRIPTION
**Pull request by jaym**
goes after #137 

TODO:
- You need to write a config file with the right values for ssl for this to work on https://localhost which is annoying. I think we can do this automatically

### CODE REVIEW

File: lib/chef/automate.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        sql_user: 'automate-cs-oc-erchef',
+        sql_cert: '/hab/svc/automate-cs-oc-erchef/config/service.crt',
+        sql_key: '/hab/svc/automate-cs-oc-erchef/config/service.key',
+        sql_rootcert: '/hab/svc/automate-cs-oc-erchef/config/root_ca.crt',
+        sql_db: 'automate-cs-oc-erchef',
+        webui_key: '/hab/svc/automate-cs-oc-erchef/data/webui_priv.pem'
+      }
+    end
+  end
+end
```
Comment (vinay033) [inline lib/chef/automate.rb:18]: add a new line here.

**Comment by vsingh-msys** [comment]
seems build is failing, Do we need any action?

**Comment by vsingh-msys** [comment]
Let me fix `simplecov` seems it versioning issue. so fixing by `gem 'simplecov', '~> 0.16.1'`

**Comment by vsingh-msys** [comment]
@tas50 & @btm This PR is now in good shape.


build is failing due to ruby dependency of chef infra.
```

Gem::RuntimeRequirementNotMetError: chef requires Ruby version >= 2.6.0. The
--
  | current ruby version is 2.5.8.224.
  | An error occurred while installing chef (16.3.45), and Bundler
  | cannot continue.
  | Make sure that `gem install chef -v '16.3.45' --source 'https://rubygems.org/'`
  | succeeds before bundling.
  |  

```

Do we need to remove expeditor builds for ruby 2.4 & 2.5?

File: spec/chef/knife/ec_key_base_spec.rb [hunk 2]
```diff
@@ -9,15 +10,16 @@ class KeyBaseTester < Chef::Knife
   let (:knife) { KeyBaseTester.new }
 
   let(:running_server_postgresql_sql_config_json) {
-    '{"private_chef": { "opscode-erchef":{}, "postgresql": { "sql_user": "jiminy", "sql_password": "secret"} }, "postgresql": { "sql_user": "jiminy", "sql_password": "secret"} }'
+    '{"private_chef": { "opscode-erchef":{}, "postgresql": { "sql_user": "jiminy", "sql_password": "secret"} } }'
```
Comment (vsingh-msys) [inline spec/chef/knife/ec_key_base_spec.rb:13]: Dup `postgresql` entry

File: lib/chef/knife/ec_key_base.rb [hunk 3]
```diff
@@ -120,7 +121,7 @@ def load_config_from_file!
                         'postgresql'
                       end
             config[:sql_user] ||= running_config['private_chef'][hash_key]['sql_user']
-            config[:sql_password] ||= sql_password
+            config[:sql_password] = (running_config['private_chef'][hash_key]['sql_password'] || sql_password) unless config[:sql_password]
```
Comment (btm) [inline lib/chef/knife/ec_key_base.rb:124]: Could keep this like the others:

```
config[:sql_password] ||= (running_config['private_chef'][hash_key]['sql_password'] || sql_password)
```

```
2.6.3 :006 > running_config = "whatever"
 => "whatever" 
2.6.3 :007 > sql_password = "secret"
 => "secret" 
2.6.3 :008 > main_config = "does not change"
 => "does not change" 
2.6.3 :009 > main_config ||= (running_config || sql_password)
 => "does not change" 
2.6.3 :010 > main_config = nil
 => nil 
2.6.3 :011 > main_config ||= (running_config || sql_password)
 => "whatever" 
2.6.3 :012 > main_config = nil
 => nil 
2.6.3 :013 > running_config = nil
 => nil 
2.6.3 :014 > main_config ||= (running_config || sql_password)
 => "secret" 
2.6.3 :015 > main_config
 => "secret" 
```

Comment (vsingh-msys) [reply lib/chef/knife/ec_key_base.rb:124]: Thanks!

**Comment by vsingh-msys** [comment]
@btm could you please have a look again and merge this PR pending for so long?

**Pull request merged**